notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+      "\u001b[0;31mUWConfigRealizeError\u001b[0m: Config could not be totally realized"
+     ]
+    }
+   ],
+   "source": [
+    "config.realize(\n",
+    "    input_config='fixtures/config/base-config.nml',\n",
+    "    update_config={\"memo\": {\"greeting\":\"Salutations\", \"recipient\":\"Mars\", \"sent\":True}},\n",
+    "    output_file='tmp/config-total.nml',\n",
+    "    total=True\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4048efdd-42f4-4400-bdba-1c8e0001d8f6",
+   "metadata": {},
+   "source": [
+    "With all values provided to fully render the config, `realize()` writes the complete config without error.\n",
+    "<!--cell 46-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "e891a446-f699-460d-b4a8-568d9d4cf631",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'memo': {'sender_id': '321',\n",
+       "  'message': 'Salutations, Mars!',\n",
+       "  'sent': True,\n",
+       "  'greeting': 'Salutations',\n",
+       "  'recipient': 'Mars',\n",
+       "  'id': 321}}"
+      ]
+     },
+     "execution_count": 24,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.realize(\n",
+    "    input_config='fixtures/config/base-config.nml',\n",
+    "    update_config={\"memo\": {\"greeting\":\"Salutations\", \"recipient\":\"Mars\", \"sent\":True, \"id\":321}},\n",
+    "    output_file='tmp/config-total.nml',\n",
+    "    total=True\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "07876d41-dad2-4c30-adba-f635050708ed",
+   "metadata": {},
+   "source": [
+    "The newly created config file is free from any unrendered Jinja2 expressions.\n",
+    "<!--cell 48-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "id": "aeab0ec5-7e6e-4309-b484-4de5dd9324b5",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "&memo\n",
+      "    sender_id = '321'\n",
+      "    message = 'Salutations, Mars!'\n",
+      "    sent = .true.\n",
+      "    greeting = 'Salutations'\n",
+      "    recipient = 'Mars'\n",
+      "    id = 321\n",
+      "/\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat tmp/config-total.nml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e121dab6-5d16-40b0-aa16-161c329a8e9a",
+   "metadata": {},
+   "source": [
+    "## Realizing Configs to a Dictionary\n",
```

Comment:
I wonder if this whole section could be minimized, just saying (paraphrasing) "This does exactly what `config.realize()` does, but only returns a `dict` value and writes no output.` In fact, the `help()` output just directs the user to see the `realize()` help, so isn't very enlightening. Maybe one small example, like the one shown below, would suffice -- especially if the suggestion I offered above, to use a `_ = config.realize(...)` assignment to suppress the `dict` output that `realize()` also returns, is accepted.

Reply:
I agree, I updated the section to remove the help call and emphasize the similarity to `realize()` sooner.